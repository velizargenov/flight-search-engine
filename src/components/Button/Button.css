input[type=button] {
  cursor: pointer;
  background: rgb(168, 168, 168);
  border: 1px solid #000000;
  border-radius: 7px;
  height: 32px;
  width: 124px;
  font-size: 12px;
  letter-spacing: 0.3px;
}

.button--search-this-flight {
  /* border: 2px solid red; */
}