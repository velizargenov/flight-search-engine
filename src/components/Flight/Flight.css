.flight {
  border: 1px solid black;
  margin: 10px 0;
  padding: 10px;
  overflow: hidden;
}
.flight figure {
  /* border:1px solid black; */
  display: inline-block;
  width: 135px;
  height: 90px;
  margin: 0;
  padding: 0;

  float: left;
}
.flight figure img {
  width: 100%;
}
.flight--price {
  /* border: 1px solid; */
  font-size: 20px;
  letter-spacing: 0.5px;
  float: left;
  position: relative;
  left: 20px;
  top: 33px;
}
.flight--details-container {
  /* border: 1px solid blue; */
  overflow: hidden;
  font-size: 11px;
  font-weight: 400;
  letter-spacing: 0.5px;
  position: relative;
  top: 10px;
  left: 20px;
  float: left;
}
.flight--details {
  /* border: 1px solid red; */
  display: inline-block;
  width: 166px;
}
.flight--details span {
  display: block;
}
.flight-details--flightNumber {
  margin-bottom: 5px;
}
.flight-details--fromTo {
  font-size: 14px;
  font-weight: 500;
  margin-bottom: 3px;
}
.flight-details--time {
  /* border: 1px solid green; */
  width: 102px;
  overflow: hidden;
  margin-bottom: 3px;
}
.flight-details--time:last-child {
  margin-bottom: 0;
}
.flight-details--time > span {
  /* border: 1px solid orange; */
  display: inline;
}
.flight-details--time > .time {
  /* border: 1px solid blue; */
  float: right;
}
.button--search-this-flight {
  float: right;
  position: relative;
  /* left: 40px; */
  top: 29px;
  
}