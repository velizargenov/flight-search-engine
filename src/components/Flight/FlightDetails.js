import React from 'react';
import PropTypes from 'prop-types';

const FlightDetails = ({ details, index }) => {
  const {
    flightNumber,
    fromAirport,
    toAirport,
    departureTime,
    arrivalTime,
  } = details[index];

  return (
    <React.Fragment>
      <div className="flight--details">
        <span className="flight-details--flightNumber">{flightNumber}</span>
        <span className="flight-details--fromTo">{fromAirport} &gt; {toAirport}</span>
        <div className="flight-details--time">
          <span>Depart:</span>
          <span className="time">{departureTime}</span>
        </div>
        <div className="flight-details--time">
          <span>Arrive:</span><span className="time">{arrivalTime}</span>
        </div>
      </div>
    </React.Fragment>
  );
};

FlightDetails.propTypes = {
  index: PropTypes.number.isRequired,
  details: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default FlightDetails;
