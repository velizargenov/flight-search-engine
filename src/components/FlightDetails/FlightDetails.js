import React from 'react';

const FlightDetails = () => (
  <h2>FlightDetails Component</h2>
);

export default FlightDetails;
