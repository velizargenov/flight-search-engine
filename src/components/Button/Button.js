import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';

const Button = ({ htmlFor, handleClick, ...otherProps }) => (
  <label htmlFor={htmlFor} className="button">
    <input
      type="button"
      id={htmlFor}
      onClick={handleClick}
      {...otherProps}
    />
  </label>
);

Button.defaultProps = {
  htmlFor: 'uniqueId',
  handleClick: () => {},
};

Button.propTypes = {
  htmlFor: PropTypes.string,
  handleClick: PropTypes.func,
};

export default Button;
