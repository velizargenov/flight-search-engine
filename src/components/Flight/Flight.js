
import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v4';

import './Flight.css';
import Button from '../Button/Button';
import FlightDetails from './FlightDetails';

const Flight = (props) => {
  const { id, airlineLogo, airlineName, details, price } = props.details;
  const { handleClicksOnSearchThisFlight } = props;

  return (
    <div className="flight">
      <figure>
        <img src={`${process.env.PUBLIC_URL}/images/${airlineLogo}`} alt={airlineName} />
      </figure>
      <div className="flight--details-container">
        {
          details.map((flight, index) => (
            <FlightDetails
              key={uuid()}
              index={index}
              details={details}
            />
          ))
        }
      </div>
      <div className="flight--price">Â£{price}</div>
      <Button
        htmlFor={id}
        handleClick={() => handleClicksOnSearchThisFlight([props.details])}
        className="button--search-this-flight"
        value="Search this flight"
      />
    </div>
  );
};

Flight.propTypes = {
  details: PropTypes.object.isRequired,
};

export default Flight;
