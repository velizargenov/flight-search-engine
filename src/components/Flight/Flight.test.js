import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Flight from './Flight';
import FlightDetails from './FlightDetails';
import Button from '../Button/Button';

jest.mock('../../mocks/flight');
const flight = require('../../mocks/flight');
const flightDetails = require('../../mocks/flight-details');

describe('Flight Component', () => {
  const wrapper = shallow(<Flight key={flight.id} details={flight} isReturnFlight={false} />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should render one Button component', () => {
    expect(wrapper.find(Button).length).toBe(1);
  });
  it('should have one wrapping <div> element', () => {
    expect(wrapper.find('div.flight').length).toBe(1);
  });
  it('should have one <figure> and one <img> element', () => {
    expect(wrapper.find('figure').length).toBe(1);
    expect(wrapper.find('img').length).toBe(1);
  });
  it('should have a container for the flight details', () => {
    expect(wrapper.find('div.flight--details-container').length).toBe(1);
  });
  it('should have a container for the total flight price', () => {
    expect(wrapper.find('div.flight--price').length).toBe(1);
  });
});

describe('FlightDetails Component', () => {
  const wrapper = shallow(<FlightDetails key={0} index={0} details={flightDetails} />);
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    expect(wrapper.length).toBe(1);
  });
  it('should have three <div> elements', () => {
    expect(wrapper.find('div').length).toBe(3);
  });
  it('should have six <span> elements', () => {
    expect(wrapper.find('span').length).toBe(6);
  });
  it('should have one wrapping <div> element', () => {
    expect(wrapper.find('div.flight--details').length).toBe(1);
  });
  it('should have details about flight number', () => {
    expect(wrapper.find('span.flight-details--flightNumber').length).toBe(1);
  });
  it('should have details about where the flight will depart', () => {
    expect(wrapper.find('span.flight-details--fromTo').length).toBe(1);
  });
  it('should have details about departure and arrival time', () => {
    expect(wrapper.find('div.flight-details--time').length).toBe(2);
  });
});
