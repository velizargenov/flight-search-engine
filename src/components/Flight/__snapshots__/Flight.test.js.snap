// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight Component matches snapshot 1`] = `
<div
  className="flight"
>
  <figure>
    <img
      alt="Air India"
      src="/images/air-india-logo.png"
    />
  </figure>
  <div
    className="flight--details-container"
  />
  <div
    className="flight--price"
  >
    Â£
  </div>
  <Button
    className="button--search-this-flight"
    handleClick={[Function]}
    htmlFor="uniqueId"
    value="Search this flight"
  />
</div>
`;

exports[`FlightDetails Component matches snapshot 1`] = `
<React.Fragment>
  <div
    className="flight--details"
  >
    <span
      className="flight-details--flightNumber"
    >
      AI-101
    </span>
    <span
      className="flight-details--fromTo"
    >
      DEL
       &gt; 
      PNQ
    </span>
    <div
      className="flight-details--time"
    >
      <span>
        Depart:
      </span>
      <span
        className="time"
      >
        10:00 AM
      </span>
    </div>
    <div
      className="flight-details--time"
    >
      <span>
        Arrive:
      </span>
      <span
        className="time"
      >
        12:10 AM
      </span>
    </div>
  </div>
</React.Fragment>
`;
